tests/test_optimize.py

```diff
@@ -205,6 +205,23 @@ def hybrid_fun_mdlist(var):
 
         assert array == pytest.approx(np.asarray(list), abs=tol)
 
+    @pytest.mark.parametrize(
+        "circuit, var", [(quant_fun, mixed_list), (quant_fun_mdarr, multid_array)]
+    )
+    def test_step_and_cost_autograd(self, bunch, circuit, var):
+        """Test that the correct cost is returned via the step_and_cost method"""
+        _, res = bunch.sgd_opt.step_and_cost(circuit, var)
```

Comment:
Do we also need to check for QNG and nesterov, since they override `step_and_cost` or `compute_grad`?

Reply:
Definitely